ProjectProtocol/project-protocol-nextjs
# OriginalPathProvider + various fixes

Fixes PRO-378

## 🛠️ Changes
OriginalPathProvider:
* stores current pathname as "prevPath" as the user clicks around
* If user enters a `redirectRoutes` namespace, prevPath is preserved as originalPath until user exits the namespace.
* Provides `getOriginalPath` function so links and forms within the "redirectRoutes" can know where to redirect to.

A major reason this wasn't working initially is because we had raw `<a />` tags in many places across the app, rather than the nextjs `Link` component. This was causing hard navigation which caused full page reloads, causing the OriginalPathProvider to lose it's state.

Also Links make navigation way snappier, so I was happy to find these <a> tags-- it accounts for some sluggishness on the site.


## 🧪 Testing
* Make sure you're logged out
* Navigate to some non-home route and click on a link to login, signup, register, etc.
* Confirm that the "X" button in the top right goes back to whichever page you came from.
* Re-initiate some login, signup, or password reset flow. Confirm that on success, you are redirected to your starting page.
